### ISSUE TITLE
Support `local` vars

### ISSUE THREAD

**Issue by ejrgilbert** [ejrgilbert/whamm#135]
# Justification #
There can be cases where you want a variable that is replicated for every matched event, but is not initialized every time the probe body is executed. This requires that you have _global_ state that is only reference-able _locally_ for a probe body.

Enter the `local` variable.

# Use Case #
Consider the following use case for this variable:
```
wasm:opcode:memory_grow {
    local int count;
    if count <= 0 {
        ...
        count = rand() % 1000; // some random number between 0 and 1000
    }
    count--;
}
```

This effectively samples the execution of each memory_grow operation using random sampling. Continuing on this idea, you could use a _global_ count to sample memory.grow at a random interval for the entire application, e.g.:
```
int count;
wasm:opcode:memory_grow {
    if count <= 0 {
        ...
        count = rand() % 1000; // some random number between 0 and 1000
    }
    count--;
}
```

# Implementation Details #
Basically implement a `report` variable that is not reported at the end of program execution.

# Make It Fast (Wizard) #

1. The JIT could put the `if count <= 0` check inline (~3 opcodes) and then call out to the probe body callback if the predicate is `true`
2. See Issue #137, the sampling logic is just a check on global state, factor this out to a dynamic check matches function.

**Comment by ejrgilbert** [comment]
New name idea: `stable`?
